### DISCUSSION TITLE
Best way to use PaddleOCR for end-to-end Key-Value Pair Extraction from ID cards

### DISCUSSION THREAD

**Discussion by atharvjairath** [PaddlePaddle/PaddleOCR#6922] [Q&A]
Hi,
I am trying to build Key-value pair Extraction from ID cards (like Driving licenses etc), where the key would be Name, Address, DOB, etc.
What would be the best way to use PaddleOCR to achieve the above?

**Comment by ThanhLV97** [comment]
You should train KIE model with your dataset.

**Comment by LDOUBLEV** [comment]
Please refer to kie algorithm: https://github.com/PaddlePaddle/PaddleOCR/blob/release%2F2.5/ppstructure/docs/kie_en.md

**Reply by ThanhLV97** [reply]
@LDOUBLEV  as my mention in previous issue. In order to onnx converting  that need paddle format but the paddle format convertingscript is maybe wrong with kie model. so How I convert kie model to onnx format using
 torchscript or something else.

**Reply by atharvjairath** [reply]
is there any way I can run a run a inference on my data using trained KIE model? @LDOUBLEV @thanh15dt2

**Reply by shahzain110** [reply]
have you found any solutions?

**Comment by AbdulDD** [comment]
you should also look into DONUT (OCR-FREE document understading) and Pix2Struct from HuggingFace.
Both are E2E models and can perform:
1. Key information extraction (extracting key and their respective values from images (like in your case))
2. Document Visual Query Answering (Doc VQA).
